pkg/productinfo/alibaba/productinfo_alibaba.go

```diff
@@ -88,16 +90,28 @@ func (e *AlibabaInfoer) getData(region string, instanceTypes []ecs.InstanceType,
 	allRegionPrices := make(map[string]map[string]productinfo.Price)
 	regionPrices := make(map[string]productinfo.Price)
 
+	var (
+		alibabaAccessKeyId     = "alibaba-access-key-id"
+		alibabaAccessKeySecret = "alibaba-access-key-secret"
+	)
+
+	testCli, _ := ecs.NewClientWithAccessKey(
+		region, viper.GetString(alibabaAccessKeyId), viper.GetString(alibabaAccessKeySecret),
+	)
+
+	request := ecs.CreateDescribeSpotPriceHistoryRequest()
+	request.RegionId = region
+	request.NetworkType = "vpc"
+	request.OSType = "linux"
+
+	log.Debugf("created new client for %s, %v", region, testCli)
 	for _, instanceType := range instanceTypes {
-		request := ecs.CreateDescribeSpotPriceHistoryRequest()
-		request.RegionId = region
-		request.NetworkType = "vpc"
-		request.InstanceType = instanceType.InstanceTypeId
-		request.OSType = "linux"
 
-		prices, err := e.client.DescribeSpotPriceHistory(request)
+		request.InstanceType = instanceType.InstanceTypeId
+		prices, err := testCli.DescribeSpotPriceHistory(request)
 		if err != nil {
-			return
+			log.Errorf("failed to get spot price history for provider [%s], region [%s], instance type [%s]. error: [%s]", "alibaba", region, instanceType, err.Error())
```

Comment:
Maybe use logrus's structural logging capabilities? `log.WithFields(....)`

Reply:
you're right, and there's an issue to improve logging (this project is small enough to be able to do it all at once), but for now we can keep it like this to be able to move forward with debugging Alibaba problems